src/test/ui/transmute/main.stderr

```diff
@@ -4,8 +4,7 @@ error[E0512]: cannot transmute between types of different sizes, or dependently-
 LL |     transmute(x)
    |     ^^^^^^^^^
    |
-   = note: source type: `<C as TypeConstructor<'a>>::T` (size can vary because of <C as TypeConstructor>::T)
-   = note: target type: `<C as TypeConstructor<'b>>::T` (size can vary because of <C as TypeConstructor>::T)
+   = note: `<C as TypeConstructor>::T` does not have a fixed size
```

Comment:
Wait, we don't allow transmuting between the same type with only lifetimes changed?
I think we do when it's beyond a pointer, even if we don't know if it's `Sized`.

Reply:
We don't allow transmutes between type projections at all it seems:

```rust
pub trait TypeConstructor {
    type T;
}

unsafe fn transmute_lifetime<C>(x: C::T)
where C: TypeConstructor {
    transmute::<C::T, C::T>(x); // ERROR !
}
```